pkg/services/ngalert/state/manager.go

```diff
@@ -385,16 +390,26 @@ func (st *Manager) staleResultsHandler(ctx context.Context, alertRule *ngModels.
 			}
 
 			if s.State == eval.Alerting {
-				st.annotateState(ctx, alertRule, s.Labels, time.Now(),
-					InstanceStateAndReason{State: eval.Normal, Reason: ""},
-					InstanceStateAndReason{State: s.State, Reason: s.StateReason})
+				previousState := InstanceStateAndReason{State: s.State, Reason: s.StateReason}
+				s.State = eval.Normal
+				s.StateReason = ngModels.StateReasonMissingSeries
+				s.EndsAt = st.clock.Now()
+				s.Resolved = true
+				s.Stale = true
```

Comment:
Do we need `Stale` if we also set `Resolved` as `true`? I'm not sure I 100% understand the purpose of this field.